mars/utils.py

```diff
@@ -1617,6 +1619,21 @@ def __str__(self):
     return new_exc_type().with_traceback(traceback)
 
 
+_func_token_cache = weakref.WeakKeyDictionary()
+
+
+def get_func_token(func):
+    try:
+        token = _func_token_cache.get(func)
+        if token is None:
+            fields = get_func_token_values(func)
+            token = tokenize(*fields)
+            _func_token_cache[func] = token
+        return token
+    except TypeError:  # cannot create weak reference to func like 'numpy.ufunc'
+        return tokenize(*get_func_token_values(func))
+
+
 def get_func_token_values(func):
```

Comment:
The `get_func_token_values` is not used by other modules except the `mars/utils.py`, it's better to rename it to `_get_func_token_values` to make it private.

Reply:
Updated